pkg/cmd/pipelinerun/delete.go

```diff
@@ -132,13 +138,18 @@ func pipelineRunLister(p cli.Params, cs *cli.Clients) func(string) ([]string, er
 	}
 }
 
-func allPipelineRunNames(p cli.Params, cs *cli.Clients) ([]string, error) {
+func allPipelineRunNames(p cli.Params, cs *cli.Clients, keep int) ([]string, error) {
 	pipelineRuns, err := cs.Tekton.TektonV1alpha1().PipelineRuns(p.Namespace()).List(metav1.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
 	var names []string
-	for _, pr := range pipelineRuns.Items {
+	var counter = 0
+	for _, pr := range prhsort.SortPipelineRunsByStartTime(pipelineRuns.Items) {
+		if keep > 0 && counter != keep {
```

Comment:
Should there be some kind of error message if keep < 0? Currently, it will delete all pipelineruns.

Reply:
in case If the user choose ?: 

--last=-1 

I guess  that's quite a weird user case

Reply:
Right, I mean it's totally unlikely, but we should prevent it by checking if keep < 0 similar to what is done for limit. So fail fast in `RunE` in this case.

Reply:
@danielhelfand isn't the code above already checking that `keep` can't be `< 0` ?